app/main.py

```diff
@@ -1,25 +1,73 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
```

Comment:
Instead of naming the protected attribute `protected_name` you could use a convention to add a prefix `_` to a variable name thus [making it protected](https://www.tutorialsteacher.com/python/public-private-protected-modifiers). Rename this attribute using conventions.